android/src/fabric/java/com/swmansion/reanimated/NativeProxy.java

```diff
@@ -1,333 +1,109 @@
-  }
-
-  @DoNotStrip
-  private void registerEventHandler(EventHandler handler) {
-    handler.mCustomEventNamesResolver = mNodesManager.getEventNameResolver();
-    mNodesManager.registerEventHandler(handler);
-  }
-
-  @DoNotStrip
-  private int registerSensor(int sensorType, int interval, SensorSetter setter) {
-    return reanimatedSensorContainer.registerSensor(
-        ReanimatedSensorType.getInstanceById(sensorType), interval, setter);
-  }
-
-  @DoNotStrip
-  private void unregisterSensor(int sensorId) {
-    reanimatedSensorContainer.unregisterSensor(sensorId);
-  }
 
-  @DoNotStrip
-  private int subscribeForKeyboardEvents(KeyboardEventDataUpdater keyboardEventDataUpdater, boolean isStatusBarTranslucent) {
-    return reanimatedKeyboardEventListener.subscribeForKeyboardEvents(keyboardEventDataUpdater, isStatusBarTranslucent);
-  }
-
-  @DoNotStrip
-  private void unsubscribeFromKeyboardEvents(int listenerId) {
-    reanimatedKeyboardEventListener.unsubscribeFromKeyboardEvents(listenerId);
-  }
-
-  public void onCatalystInstanceDestroy() {
-    mScheduler.deactivate();
-    mHybridData.resetNative();
-  }
-
-  public void prepare(LayoutAnimations LayoutAnimations) {
-    if (Utils.isChromeDebugger) {
-      Log.w("[REANIMATED]", "You can not use LayoutAnimation with enabled Chrome Debugger");
-      return;
+    // TODO: move to NativeProxyCommon.java when layout animations are fixed on Fabric 
+    public void prepareLayoutAnimations(LayoutAnimations LayoutAnimations) {
+        if (Utils.isChromeDebugger) {
+            Log.w("[REANIMATED]", "You can not use LayoutAnimation with enabled Chrome Debugger");
+            return;
+        }
+        mNodesManager = mContext.get().getNativeModule(ReanimatedModule.class).getNodesManager();
+
+        AnimationsManager animationsManager =
+                mContext
+                        .get()
+                        .getNativeModule(ReanimatedModule.class)
+                        .getNodesManager()
+                        .getAnimationsManager();
+
+        WeakReference<LayoutAnimations> weakLayoutAnimations = new WeakReference<>(LayoutAnimations);
+        animationsManager.setNativeMethods(
+                new NativeMethodsHolder() {
+                    @Override
+                    public void startAnimation(int tag, String type, HashMap<String, Float> values) {}
+
+                    @Override
+                    public boolean isLayoutAnimationEnabled() {
+                        return false;
+                    }
+
+                    @Override
+                    public int findPrecedingViewTagForTransition(int tag) {
+                        return -1;
+                    }
+
+                    @Override
+                    public boolean hasAnimation(int tag, String type) {
+                        return false;
+                    }
+
+                    @Override
+                    public void clearAnimationConfig(int tag) {}
+                });
     }
```

Comment:
I think the only difference between these two methods is the object passed to `animationsManager.setNativeMethods` call.

Can we add a new method `createNativeMethodsHolder` that returns this complex object, separately for Paper and Fabric, and move `prepareLayoutAnimations` to `NativeProxyCommon.java`?